scripts/prepare_dolly.py

```diff
@@ -0,0 +1,170 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import sys
+from pathlib import Path
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+import torch
+import requests
+import json
+from torch.utils.data import random_split
+from lit_gpt.tokenizer import Tokenizer
+from tqdm import tqdm
+
+
+DESTINATION_PATH = Path("data/dolly")
+CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+DATA_FILE_URL = "https://huggingface.co/datasets/databricks/databricks-dolly-15k/resolve/main/databricks-dolly-15k.jsonl"
+DATA_FILE_NAME = "dolly_data_cleaned.json"
+MASK_INPUTS = False
+TEST_SPLIT_FRACTION = 0.1333  # to get exactly 2000 test samples
+SEED = 42
+IGNORE_INDEX = -1
+
+
+def prepare(
+    destination_path: Path = DESTINATION_PATH,
+    checkpoint_dir: Path = CHECKPOINT_DIR,
+    test_split_fraction: float = TEST_SPLIT_FRACTION,
+    seed: int = SEED,
+    mask_inputs: bool = MASK_INPUTS,
+    data_file_name: str = DATA_FILE_NAME,
+    data_file_url: str = DATA_FILE_URL,
+    ignore_index: int = IGNORE_INDEX,
+) -> None:
+    """Prepare the Alpaca dataset for instruction tuning.
+
+    The output is a training and test dataset saved as `train.pt` and `test.pt`,
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+
+    with open(checkpoint_dir / "lit_config.json", "r") as file:
+        config = json.load(file)
+        max_seq_length = config["block_size"]
+
+
+    destination_path.mkdir(parents=True, exist_ok=True)
+    data_file_path = destination_path / data_file_name
+    print("Loading data file...")
+    download_if_missing(data_file_path, data_file_url)
+    
+    with open(data_file_path, "r") as file:
```

Comment:
Windows users will need 
```suggestion
    with open(data_file_path, "r", encosing="utf-8") as file:
```